design-documents/frontend/admin-guidelines.md

```diff
@@ -0,0 +1,194 @@
+# Magento Admin Guidelines
+
+## Problem
+
+There are already a handful of React applications built for Magento Admin use, but they have different dependencies and versions between them and we don't yet have a micro front-end architecture to manage them.
+
+## Solution (for now)
+
+We must do our best to align common dependencies manually.
+
+The following [`package.json`](#packagejson) includes a list of dependencies that you should use instead of their alternatives. Please also stick within the version ranges specified to support a graceful transition to a future architecture.
+
+To aid in the process of using these dependencies, consider using the [TSX Yeoman Generator](https://www.npmjs.com/package/generator-tsx).
+
+For more information behind these technology choices, refer to the [Technical Vision](./technical-vision.md).
+
+## package.json
+
+Each section of the `package.json` file has been broken down into different topics.
+
+- [version](#version)
+- [scripts](#scripts)
+- [browserslist](#browserslist)
+- [dependencies](#dependencies)
+  - [Utility functions](#utility-functions)
+  - [SSR](#ssr)
+  - [React](#react)
+  - [State management](#state-management)
+  - [Code splitting](#code-splitting)
+  - [i18n](#i18n)
+  - [Theming](#theming)
+  - [Relay](#relay)
+  - [TypeScript import helpers](#typescript-import-helpers)
+- [devDependencies](#devdependencies)
+  - [@types](#types)
+  - [Redux mock store](#redux-mock-store)
+  - [Testing library](#testing-library)
+
+### version
+
+The version of this `package.json` follows [semver](https://semver.org/) and will change with this document accordingly.
+
+```json
+"version": "0.1.0",
+```
+
+### scripts
+
+Customize your CRA configuration with [craco][].
+
+- [Don't eject](https://create-react-app.dev/docs/alternatives-to-ejecting).
+- Avoid forking, because you can oftentimes do the same thing by creating your own [craco][] plugin. See [craco plugins on npm](https://www.npmjs.com/search?q=craco%20plugin) and the [craco recipes](https://github.com/sharegate/craco/tree/master/recipes) for guidance.
+
+```json
+"scripts": {
+  "start": "craco start",
+  "build": "craco build",
+  "test": "craco test"
+},
+```
+
+### [browserslist](https://github.com/browserslist/browserslist#readme)
+
+Don't forget your [polyfills](https://create-react-app.dev/docs/supported-browsers-features#supported-language-features)!
+
+```json
+"browserslist": [
+  ">0.2%",
+  "not dead",
+  "not ie <= 11",
+  "not op_mini all"
+],
+```
+
+### dependencies
+
+#### Utility functions
+
+Use vanilla (native) JS if possible, utility functions if not. [You don't (may not) need Lodash/Underscore](https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#you-dont-may-not-need-lodashunderscore-). See also [just](https://github.com/angus-c/just#just).
+
+```json
+"@queso/camel-case": "^0",
+"@queso/kebab-case": "^1",
+```
+
+#### SSR
```

Comment:
[Next.js](https://nextjs.org/) recommends the use of [isomorphic-unfetch](https://www.npmjs.com/package/isomorphic-unfetch) for universal [data fetching](https://nextjs.org/docs#fetching-data-and-component-lifecycle). Might be another option to consider as it works very similar to the built in `fetch`.

Reply:
Yep. I've been down this 🐰 🕳 a few times now and I keep coming to the same conclusion. Take a look at [ky's usage[(https://www.npmjs.com/package/ky#usage) compared to plain fetch. Much more [readable and expressive](https://github.com/jedmao/tsx/blob/e7fb79851f217c7a05af91ba44368cb363e44d2d/src/config/environment.ts#L12-L19) and comes with 1st-class TypeScript support to boot! Happy to open another conversation on the matter though if you have some arguments to present.